File: 101/exercise_rework/easy_6/combining_arrays.rb
# combining_arrays.rb

def merge(arr, arr1)
  arr | arr1
end

def keep_iterating?(arr, arr1, idx)
  arr.fetch(idx, 'nope') != 'nope' || arr1.fetch(idx, 'nope') != 'nope'
end

def fe_merge(arr, arr1)
  new_arr = []
  idx = 0
  while keep_iterating?(arr, arr1, idx)
    element = arr.fetch(idx, 'nope') unless arr.fetch(idx, 'nope') == 'nope'
    element1 = arr1.fetch(idx, 'nope') unless arr1.fetch(idx, 'nope') == 'nope'
    new_arr << element unless new_arr.include?(element)
    new_arr << element1 unless new_arr.include?(element1)
    idx += 1
  end
  new_arr.sort
end

File: 210/js_small_problems/easy_4/combining_arrays.js
// combining_arrays.js

function union(arr, arr1) {
  var unionArray = [];

  arr.forEach( function(el) {
    if (!unionArray.includes(el)) {
      unionArray.push(el);
    }
  });

  arr1.forEach( function(el) {
    if (!unionArray.includes(el)) {
      unionArray.push(el);
    }
  });
  return unionArray;
}

console.log(union([1, 3, 5], [3, 6, 9]));    // [1, 3, 5, 6, 9]